### PR TITLE
Add application error codes

### LINKED ISSUE #45: Split application and transport error spaces
This manifests in two ways:

1. Reset takes minq.ErrorCode, but it probably wants a new type minq.ApplicationError (uint16)
2. Shutdown doesn't take an argument

### PR DESCRIPTION
**Pull request by martinthomson**
Closes #45.

### CODE REVIEW

File: stream.go [hunk 1]
```diff
@@ -100,7 +100,7 @@ type hasIdentity interface {
 
 type sendStreamMethods interface {
```
Comment (ekr) [inline stream.go]: Why did you change these to uint16 from ErrorCode?

Comment (martinthomson) [reply stream.go]: Because minq.ErrorCode is an enum that minq owns.  This is supposed to be a 16-bit slot for applications to do with as they please.  kQuicErrorNoError might appear to be a valid value for ErrorCode, but it is misleading because that codepoint is assigned to STOPPING.  All other codes are application-defined.
